time_calculation_LLM.py

```diff
@@ -1304,97 +1653,105 @@ def _prepare_execution_graphs(
 
         comm_metadata = self._build_comm_metadata(
             reduction_sizes=reduction_sizes,
-            local_comp=local_comp,
+            local_comp=local_comp, #when doing all_reduce between dp nodes, there is local computation on each node to apply gradients
             embedding_size=embedding_size,
             softmax_size=softmax_size,
             cross_layer_bytes=cross_layer_bytes,
         )
 
-        shapes = gemm_shapes or LLM_util.process_gemm_shapes(
-            batch_size,
-            seq_len,
-            hidden_dim,
-            num_heads,
-            ffn_dim,
-            vocab_size,
-            option="multiply_batch_into_m",
-        )
-
+        transformer_operation_entries: List[Dict[str, Any]] = []
         transformer_comm_metadata: Dict[str, Dict[str, Any]] = {}
-        transformer_gemm_entries: List[Dict[str, Any]] = []
-
-        gemm_shapes = {
-            "qkv_proj": shapes["qkv_proj"],
-            "attention_score": shapes["attention_score"],
-            "attention_output": shapes["attention_output"],
-            "output_proj": shapes["output_proj"],
-            "ffn1": shapes["ffn1"],
-            "ffn2": shapes["ffn2"],
-        }
-        gemm_ops = tuple(gemm_shapes.keys())
-
-        op_sequence = (
-            "layernorm1",
-            "qkv_proj",
-            "attention_score",
-            "attention_scale_softmax",
-            "attention_output",
-            "output_proj",
-            "residual1",
-            "ffn1",
-            "ffn2",
-            "residual2",
-            "layernorm2",
-        )
+        parallelism_mode = self.get_parallelism_mode()
+        if parallelism_mode == "cp" or parallelism_mode == "tp-cp":
+            for key in ("layernorm1", "qkv_proj", "attention_score",  "attention_scale_softmax", "attention_output", "output_proj", "layernorm2", "MLP"):
```

Comment:
One way to do this is to make a global var that includes all of these names. So at the very least it's easy to understand what this for loop is doing. For example:

GEMM_OPS = ("qkv_proj",..)

for op in GEMM_OPS:
	...
	
The above example is the same as what you have but makes it easy for the reader to understand that GEMM OPS is what we are itearting over, not an arbitrary list. ALternatively put an infomative comment.